### PR TITLE
Bugfix: clear FCC store before updating

### PR DESCRIPTION
**Pull request by paullatzelsperger**
## What this PR changes/adds

This PR cleans out the FCC cache store before re-populating it again with the results of a crawl run.

## Why it does that

To avoid having stale entries: when a connector removes a contract definition, all other connectors would still have corresponding (but invalid) offers.

## Further notes
n/a

## Checklist

- [x] added appropriate tests?
- [x] performed checkstyle check locally?
- [x] added/updated copyright headers?
- [x] documented public classes/methods?
- [ ] added/updated relevant documentation?
- [x] added relevant details to the changelog? (_skip with label `no-changelog`_)
- [x] formatted title correctly? (_take a look at the [CONTRIBUTING](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/main/CONTRIBUTING.md#submit-a-pull-request) and [styleguide](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/main/styleguide.md) for details_)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -35,12 +35,15 @@ in the detailed section referring to by linking pull requests or issues.
...
 * Extract single `PolicyArchive` implementation (#1158)
-* Replace `accessPolicy` and `contractPolicy` with `accessPolicyId` and `contractPolicyId` on `ContractDefinition` (#1144)
+* Replace `accessPolicy` and `contractPolicy` with `accessPolicyId` and `contractPolicyId` on `ContractDefinition` (
+  # 1144)
 * Replace `policy` with `policyId` on `ContractAgreement` (#1220)
 * All DMgmt Api methods now produce and consume `APPLICATION_JSON` (#1175)
 * Make data-plane public api controller asynchronous (#1228)
 * Provide In-mem implementations by default (#1130)
 * Changed Catalog config keys and switched from minutes to seconds
+* Usage of `NooTransactionContext` in (SQL-)Tests (#1119)
```
Comment (marcgs) [inline CHANGELOG.md:45]:
```suggestion
* Usage of `NoopTransactionContext` in (SQL-)Tests (#1119)
```

File: extensions/catalog/federated-catalog-cache/src/test/java/org/eclipse/dataspaceconnector/catalog/defaults/store/InMemoryFederatedCacheStoreTest.java [hunk 2]
```diff
@@ -110,4 +113,23 @@ void queryCacheContainingTwoDistinctAssets_shouldReturnBothContractOffers() {
                 .anySatisfy(co -> assertThat(co.getAsset().getId()).isEqualTo(assetId1))
                 .anySatisfy(co -> assertThat(co.getAsset().getId()).isEqualTo(assetId2));
     }
+
+    @Test
+    void deleteAll() {
```
Comment (marcgs) [inline extensions/catalog/federated-catalog-cache/src/test/java/org/eclipse/dataspaceconnector/catalog/defaults/store/InMemoryFederatedCacheStoreTest.java:118]: @paullatzelsperger do you think it would be worth writing integration tests for the federated catalog logic? Probably in a separate story as there are none at the moment AFAIS.

**Review by marcgs — APPROVED** [review]

File: extensions/catalog/federated-catalog-cache/src/test/java/org/eclipse/dataspaceconnector/catalog/defaults/store/InMemoryFederatedCacheStoreTest.java (hunk 2 above)
Comment (paullatzelsperger) [reply extensions/catalog/federated-catalog-cache/src/test/java/org/eclipse/dataspaceconnector/catalog/defaults/store/InMemoryFederatedCacheStoreTest.java:118]: @marcgs yes that is likely a good idea :)
however, I wanted to get this in quickly, and i-tests or e2e-tests for the catalog are no small feat.

**Comment by bscholtes1A** [comment]
Hi @paullatzelsperger 
what happens if one node is not available in the current crawl run? (temporary unavailability)
If my understanding is correct, then it means we will lose the assets proposed by this node, thus we lose a bit of the "caching" aspect.  WDYT?

**Comment by paullatzelsperger** [comment]
> Hi @paullatzelsperger what happens if one node is not available in the current crawl run? (temporary unavailability) If my understanding is correct, then it means we will lose the assets proposed by this node, thus we lose a bit of the "caching" aspect. WDYT?

@bscholtes1A If the request comes back unsuccessful, the loader would not neither clean nor update the cache, so the dirty items remain. even if we chose to clear the cache before receiving the response, those assets would "re-appear" on the next run. as we now do not have a standalone catalog runtime yet, the catalog being down would also mean the connector is down, so no negotiation or transfer would be possible anyway. 

TL;DR: I don't think it's a problem :)

File: extensions/catalog/federated-catalog-cache/src/test/java/org/eclipse/dataspaceconnector/catalog/defaults/store/InMemoryFederatedCacheStoreTest.java (hunk 2 above)
Comment (ndr-brt) [reply extensions/catalog/federated-catalog-cache/src/test/java/org/eclipse/dataspaceconnector/catalog/defaults/store/InMemoryFederatedCacheStoreTest.java:118]: Created an issue about: https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/issues/1279

**Review by bscholtes1A — APPROVED** [review]
